### PR TITLE
Clean up and improve comments for spark extensions

### PR DESCRIPTION
**Pull request by pan3793**
# :mag: Description

This pull request 

- improves comments for SPARK-33832
- removes unused `spark.sql.analyzer.classification.enabled` (I didn't update the migration rules because this configuration seems never to work properly)

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Test Plan 🧪

Review

---

# Checklist 📝

- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

**Comment by pan3793** [comment]
cc @ulysses-you, do we still require the Kyuubi customized `InsertShuffleNodeBeforeJoin` after SPARK-33832, seems SPARK-33832 is already smart enough

**Comment by wForget** [comment]
> cc @ulysses-you, do we still require the Kyuubi customized `InsertShuffleNodeBeforeJoin` after SPARK-33832, seems SPARK-33832 is already smart enough

I think we may still need `InsertShuffleNodeBeforeJoin`, which can also be used to optimize multiple table join skew situations.

**Comment by pan3793** [comment]
@wForget I see, they are handling different things, SPARK-33832 for skew optimization after Join, but this one is for before Join

File: extensions/spark/kyuubi-extension-spark-3-4/src/main/scala/org/apache/kyuubi/sql/InsertShuffleNodeBeforeJoin.scala [hunk 1]
```diff
@@ -49,7 +49,7 @@ object InsertShuffleNodeBeforeJoin extends Rule[SparkPlan] {
     }
   }
 
-  // Since spark 3.3, insertShuffleBeforeJoin shouldn't be applied if join is skewed.
+  // SPARK-33832 (Spark 3.3) already handles skew join cases
```
Comment (wForget) [inline extensions/spark/kyuubi-extension-spark-3-4/src/main/scala/org/apache/kyuubi/sql/InsertShuffleNodeBeforeJoin.scala:52]: As https://github.com/apache/kyuubi/pull/6260#issuecomment-2041317569 commented, should we revert this change?

Comment (pan3793) [reply extensions/spark/kyuubi-extension-spark-3-4/src/main/scala/org/apache/kyuubi/sql/InsertShuffleNodeBeforeJoin.scala:52]: yes, the comment is incorrect then. but seems SPARK-46170 recovers the skew join cases

**Review by wForget — APPROVED** [review]
Thanks, LGTM.

**Comment by pan3793** [comment]
Merged to master/1.9
